### PR TITLE
Add default Data Collected About value to Occurrence Point Forms

### PR DESCRIPTION
**Pull request by martha**
## _Merging this PR_
- use the squash-merge strategy for PRs targeting a release-X branch

## Description
Bugfix arising from https://github.com/open-path/Green-River/issues/6972#issuecomment-2520547404

## Type of change
[//]: # (e.g., Bug fix, New feature, Code clean-up, Dependency update)
Bug fix

## Checklist before requesting review
[//]: # (Remove any items that are not applicable)
- [ ] I have performed a self-review of my code
- [ ] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [ ] If adding a new endpoint / exposing data in a new way, I have:
  - [ ] ensured the API can't leak data from other data sources
  - [ ] ensured this does not introduce N+1s
  - [ ] ensured permissions and visibility checks are performed in the right places
- [ ] I have updated the documentation (or not applicable)
- [ ] I have added spec tests (or not applicable)
- [ ] I have provided testing instructions in this PR or the related issue (or not applicable)

[//]: # NOTE: system tests may fail if there is no branch on the hmis-frontend that matches the Source  or Target branch of this PR. This is expected

### CODE REVIEW

File: drivers/hmis/app/graphql/types/hmis_schema/occurrence_point_form.rb [hunk 1]
```diff
@@ -27,6 +27,10 @@ def definition(parent:)
       definition
     end
 
+    def data_collected_about
```
Comment (martha) [inline drivers/hmis/app/graphql/types/hmis_schema/occurrence_point_form.rb]: @gigxz , I'm still a bit confused about what caused the bug. I wasn't able to reproduce it locally, and also noticed that in the spec test, the instance has a nil value for data collected about. https://github.com/greenriver/hmis-warehouse/blob/release-143/drivers/hmis/spec/requests/hmis/enrollment_occurrence_point_spec.rb#L69

Comment (martha) [reply drivers/hmis/app/graphql/types/hmis_schema/occurrence_point_form.rb]: I can keep investigating before merging this PR if you think it is non-urgent.

Comment (martha) [reply drivers/hmis/app/graphql/types/hmis_schema/occurrence_point_form.rb]: Enrollment occurrence_point_forms provides a default https://github.com/greenriver/hmis-warehouse/blob/release-143/drivers/hmis/app/models/hmis/hud/enrollment.rb#L360

And so do Client Detail Forms https://github.com/greenriver/hmis-warehouse/blob/release-143/drivers/hmis/app/graphql/types/hmis_schema/query_type.rb#L512

Project occurrence_point_form_instances don't appear to provide a default https://github.com/greenriver/hmis-warehouse/blob/release-143/drivers/hmis/app/models/hmis/hud/project.rb#L297-L309

But this is no longer used from the frontend, and the Sentry error refers to `GetEnrollmentDetails`

Comment (gigxz) [reply drivers/hmis/app/graphql/types/hmis_schema/occurrence_point_form.rb]: > Project occurrence_point_form_instances don't appear to provide a default .. But this is no longer used from the frontend, and the Sentry error refers to GetEnrollmentDetails

I see, thanks for that summary. At the time when the issue surfaced, the backend was on release-142 and the frontend was still on release-141. So the frontend was still using the project occurrence point details query in GetEnrollmentDetails. 

So now that the frontend and backends are all on 142, this bug would only be present when resolving `project { occurrencePointForms }`, which we no longer do. It's still part of the API, though. I think its safest to merge this change.

File: drivers/hmis/app/graphql/types/hmis_schema/occurrence_point_form.rb [hunk 2]
```diff
@@ -27,6 +27,10 @@ def definition(parent:)
...
+      object.data_collected_about || 'ALL_CLIENTS'
+    end
+
```
Comment (gigxz) [inline drivers/hmis/app/graphql/types/hmis_schema/occurrence_point_form.rb]: There's a more succinct way to set the default:
```
field :data_collected_about, Types::Forms::Enums::DataCollectedAbout, null: false, default_value: 'ALL_CLIENTS'
```

`default_value` is not part of ruby-graphql, we defined it ourselves on BaseField: https://github.com/greenriver/hmis-warehouse/blob/30ba0e4dd283de038b2bafb08c9fd1220fbce175/drivers/hmis/app/graphql/types/base_field.rb#L71-L81

**Review by gigxz — APPROVED** [review]

File: drivers/hmis/app/graphql/types/hmis_schema/occurrence_point_form.rb (hunk 1 above)
Comment (martha) [reply drivers/hmis/app/graphql/types/hmis_schema/occurrence_point_form.rb]: Ah, it makes sense that this is related to the backend and frontend being on different releases. I was wondering if it was something like that. I was able to reproduce it and confirm that setting the default value fixes the problem.

Comment (martha) [reply drivers/hmis/app/graphql/types/hmis_schema/occurrence_point_form.rb]: Is it ok to keep this PR targeted at release-143, or would you like me to open it against release-142?

Comment (gigxz) [reply drivers/hmis/app/graphql/types/hmis_schema/occurrence_point_form.rb]: I think 143 is good, thanks!
